src/lavinmq/mqtt/client.cr

```diff
@@ -52,7 +52,7 @@ module LavinMQ
       private def read_loop
         socket = @socket
         if socket.responds_to?(:"read_timeout=")
-          socket.read_timeout = (@keepalive * 1.5).seconds # 1.5 according to [MQTT-3.1.2-24].
+          socket.read_timeout = @keepalive.zero? ? nil : (@keepalive * 1.5).seconds
```

Comment:
I think it would be good to keep the comment :)
```suggestion
          # 1.5 times grace period according to [MQTT-3.1.2-24].
          socket.read_timeout = @keepalive.zero? ? nil : (@keepalive * 1.5).seconds
```